js/ui/number_box/number_box.mask.js

```diff
@@ -496,17 +496,23 @@ var NumberBoxMask = NumberBoxBase.inherit({
             return;
         }
 
-        var caret = this._caret();
+        var caret = this._caret(),
+            isIeMinusKey = e.key === NUMPUD_MINUS_KEY_IE;
+
         if(caret.start !== caret.end) {
-            this._caret(maskCaret.getCaretInBoundaries(0, this._getInputVal(), this._getFormatPattern()));
+            if(e.key === MINUS || isIeMinusKey) {
+                this.reset();
```

Comment:
What about case when we have selected a part of the number?

Reply:
We also have a breaking change here. When we call the `reset` method, the value will be changed but when we press `minus` button without selection, the value will not be changed.

The value should not be changed before focus out or change event. It should be possible to cancel changing with `esc` before the change event